### DISCUSSION TITLE
QNAP Docker Install : Error pinging Docker server

### DISCUSSION THREAD

**Discussion by CrownParkComputing** [coder/coder#6587] [General] [answered]
So I have Coder setup as my docker instance and can connect etc, I cannot launch any docker templates as I get the following error 

Error: Error pinging Docker server: Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/_ping": dial unix /var/run/docker.sock: connect: permission denied

I have no issues with Yacht, portainer or any other docker images creating new docker containers etc. 
When I connect to coder-1 and run this command i get unable to connect 
curl --unix-socket /run/docker.sock  http://localhost/_ping 
Any help please ?

**Comment by jianjustin** [comment]
I have same question in mac m1

**Comment by ericpaulsen** [comment]
I've reproduced this on my M2 mac. `docker context ls` returns the expected `docker.sock` location:

```
NAME                TYPE                DESCRIPTION                               DOCKER ENDPOINT                              KUBERNETES ENDPOINT   ORCHESTRATOR
default *           moby                Current DOCKER_HOST based configuration   unix:///var/run/docker.sock
desktop-linux       moby  
```

**Reply by rajasaur** [reply]
You need to add the `coder` user also to the docker group and then do a `newgrp docker` to refresh the changes.

If you already have a local `coder server` running, you will need to restart that process too as that assumes that coder is NOT in the docker group until the above is done.

**Comment by phorcys420** [comment, marked as answer]
The `coder` user has to be part of the `docker` group to access the Docker socket, otherwise it can't communicate with the Docker daemon.
e.g `usermod -aG docker coder` if running standalone, you might also need to run `newgrp coder` or reboot to refresh the groups

---

When running Coder within Docker, see the following docs on what values to pass to the Docker CLI or to edit in the docker-compose.yml file.
https://coder.com/docs/install/docker
